### PR TITLE
tests: add cypress-axe test for homepage

### LINKED ISSUE #4361: Add accessibility/aXe audits on the website as part of CI
Accessibility is important! 🦾 💖 

I'm going to:
1. Run aXe on all the website pages
2. Send a PR to fix any issues
3. Integrate aXe into the build, such as with cypress

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
## PR Checklist

-   [x] Addresses an existing issue: fixes #4361
-   [x] That issue was marked as [accepting prs](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aopen+is%3Aissue+label%3A%22accepting+prs%22)
-   [x] Steps in [CONTRIBUTING.md](https://github.com/typescript-eslint/typescript-eslint/blob/main/CONTRIBUTING.md) were taken

## Overview

Uses the failures table logging [suggested in cypress-axe's docs](https://github.com/component-driven/cypress-axe#using-the-violationcallback-argument): 

```
┌─────────┬──────────────────────────────────────────────────────┬───────────────────────┬────────────┬───────┐
│ (index) │                       description                    │          id           │   impact   │ nodes │
├─────────┼──────────────────────────────────────────────────────┼───────────────────────┼────────────┼───────┤
│    0    │  'Ensure image alternative is not repeated as text'  │ 'image-redundant-alt' │  'minor'   │   1   │
│    1    │ 'Ensures all page content is contained by landmarks' │       'region'        │ 'moderate' │   1   │
└─────────┴───────────────────────────────────────────────────────────────────────────────────────────────────┴
```

Screenshot and video artifacts are made available per https://docs.github.com/en/actions/managing-workflow-runs/downloading-workflow-artifacts.

Index -- is accessible (failed)

### CODE REVIEW

File: packages/website/src/components/config/ConfigEditor.module.css [hunk 1]
```diff
@@ -1,3 +1,7 @@
+.DocSearch-Button-Placeholder {
+  color: darkorchid;
+}
```
Comment (JoshuaKGoldberg) [inline packages/website/src/components/config/ConfigEditor.module.css:3]: Was playing around with this, will likely remove...

**Review by bradzacher — APPROVED** [review]

**Comment by bradzacher** [comment]
image

awkward

**Comment by JoshuaKGoldberg** [comment]
Better we get the e2e test flake out now than in random unrelated PRs...

**Review by bradzacher — APPROVED** [review]

**Comment by armano2** [comment]
i noticed somethign bad happening after this change, 

`html:root` has higher priority than `html[data-theme='dark']` and it looks that dark mode overrides are no longer working correctly

image

and this should be 

image

**Pull request merged**